### ISSUE TITLE
AtlasCli Config Root Should Be A Path

### ISSUE THREAD

**Issue by tpetracca** [palantir/atlasdb#587]
I don't think a single word configuration root that we recursively scan the yaml file for is good enough in case the config root conflicts with something else (or if a file actually has multiple atlas configurations for whatever reason).  It should really be a full qualified path.  So in the following example it would be `root.branch_1.atlasdb` or `/root/branch_1/atlasdb`

```
root:
  branch_0:
    atlasdb:
      this_is_not: an_atlas_config
  branch_1:
    atlasdb:
      keyValueService:
        <rest-of-legit-AtlasDbConfig>
```

**Comment by tpetracca** [comment]
@rjullman @gbrova (I'm assuming this would eventually get kicked to one of the two of you).  I already have an implementation for this that's a part of my soon to be large PR for getting live good backup/restore running at a deployment.  Mostly I was curious if people had thoughts on the '.' vs. '/' models.  The yaml library we use, it's library naturally takes in the '/' form but I'm indifferent and can do the parsing before we hand it to it to do the '.' version.

And the real reason I needed this was because my work around to not having atlas-cli currently deployable with the products that consume it mean I deploy an "atlasdb-cli-wrapper" distribution whose configuration file contains the atlas configuration for every atlas service deployed on that same host and you pass the path to the correct config to use when executing the cli.

**Comment by joelea** [comment]
Entirely agree with this, was just thinking that this was going to cause us problems with one potential solution to some configuration problems.

**Comment by tpetracca** [comment]
This was addressed by #646
